lib/PrintAsClang/PrintClangFunction.cpp

```diff
@@ -713,6 +713,18 @@ ClangRepresentation DeclAndTypeClangFunctionPrinter::printFunctionSignature(
     if (kind == FunctionSignatureKind::CxxInlineThunk)
       ClangSyntaxPrinter(os).printGenericSignature(Signature);
   }
+  if (const auto *enumDecl = FD->getDeclContext()->getSelfEnumDecl())
+  {
+    // We cannot emit functions with the same name as an enum case yet, the resulting header
+    // does not compiler.
+    // FIXME: either do not emit cases as inline members, or rename the cases or the
+    //        colliding functions.
+    for (const auto *enumElement : enumDecl->getAllElements()) {
+      auto elementName = enumElement->getName();
+      if (!elementName.isSpecial() && elementName.getBaseIdentifier().str() == name)
+        return ClangRepresentation::unsupported;
```

Comment:
Unfortunately, with the current approach, we sometimes already write some strings to the ostream (that writes to the header) by the time we realise that a construct is unsupported. In this particular case we only added two spaces, so it should not cause any trouble (other than formatting).

There are other code paths below returning unsupported after the declaration is already partially written to the stream which is bad. I think this should be fixed independently of this PR, probably by writing the declaration to a temporary buffer, and commit/write that buffer to the output stream at the very end.